Pocket/pocket-ios
# feat(sentry): add user feedback reporting for errors

## Summary
Add custom UI for user to report an issue that gets sent to Sentry and populates the user feedback section. This custom report an issue view can be accessed through searches current error views (banner and empty state). 

## References 
IN-1162

## Implementation Details
Create new UI for reporting an issue `ReportIssueView` that utilizes the `ReportField` component. Added a parameter for our `PocketButtonStyle` to consider `small` button types. This is needed to add a small button to our banner view. This is a workaround, but when we refactor buttons we should be using the font size and padding to create over static button height (up to debate). Also, added a `buttonType` to differentiate a `premium` vs. `normal` button to be used in our empty states.

Also, added a new method `sentryCaptureUserFeedback` in our `Log.swift` file to call Sentry API for user feedback.

## Test Steps
1. Set up environment to return error message (whether through modifying code or using Proxyman)
2. Tap on send a report button
3. Fill out report form and verify that email is populated
4. Verify that event is sent to Sentry

## PR Checklist:

## Screenshots
| Light Mode | Dark Mode | 
| ---- | ---- | 
| ![Simulator Screenshot - iPhone 14 - 2023-05-25 at 12 12 18](https://github.com/Pocket/pocket-ios/assets/6743397/401a40e0-1703-437b-a989-66d4538a7c8c) |  ![Simulator Screenshot - iPhone 14 - 2023-05-25 at 12 12 08](https://github.com/Pocket/pocket-ios/assets/6743397/1ce88878-93e9-43ec-9fd9-105ce2ca1d5b) |

| Banner Mode | Empty State Mode | 
| ---- | ---- | 
| ![simulator_screenshot_5FF5052C-1544-48F2-B791-F6670F3B772E](https://github.com/Pocket/pocket-ios/assets/6743397/07a51e22-3463-4ec2-9bdd-009a292953fe) | ![simulator_screenshot_90617B28-5373-4D8E-98D5-F1A48C58D1B1](https://github.com/Pocket/pocket-ios/assets/6743397/abe3e8e3-dce5-4637-b836-6647bfd081c6) |
